modules/auxiliary/gather/mongodb_ops_manager_diagnostic_archive_info.rb

```diff
@@ -0,0 +1,212 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'digest/md5'
+require 'zlib'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'MongoDB Ops Manager Diagnostic Archive Sensitive Information Retriever',
+        'Description' => %q{
+          MongoDB Ops Manager Diagnostics Archive does not redact SAML SSL Pem Key File Password
+          field (mms.saml.ssl.PEMKeyFilePassword) within app settings. Archives do not include
+          the PEM files themselves. This module extracts that unredacted password and stores
+          the diagnostic archive for additional manual review.
+
+          This issue affects MongoDB Ops Manager v5.0 prior to 5.0.21 and
+          MongoDB Ops Manager v6.0 prior to 6.0.12.
+
+          API credentials with the role of GLOBAL_MONITORING_ADMIN or GLOBAL_OWNER are required.
+
+          Successfully tested against MongoDB Ops Manager v6.0.11.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/advisories/GHSA-xqvf-v5jg-pxc2'],
+          [ 'URL', 'https://www.mongodb.com/docs/ops-manager/current/reference/configuration/#mongodb-setting-mms.https.PEMKeyFilePassword'],
+          [ 'CVE', '2023-0342']
+        ],
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'DisclosureDate' => '2023-06-09',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptString.new('API_USERNAME', [ true, 'User to login with for API requests', '']),
+        OptString.new('API_PASSWORD', [ true, 'Password to login with for API requests', '']),
```

Comment:
I'm wondering if these options should match the corresponding name used by the application: `Public Key` and `Private Key`